qiskit/test/mock/utils/backend.py

```diff
@@ -0,0 +1,308 @@
+                                  sequence=measure_command_sequence)
+
+        cmd_def = [measure_command]
+
+        for i in range(self.n_qubits):
+            cmd_def += [
+                Command(name='u1', qubits=[i],
+                        sequence=[PulseQobjInstruction(name='fc', ch='d{}'.format(i),
+                                                       t0=0, phase='-P0')]),
+                Command(name='u2', qubits=[i],
+                        sequence=[PulseQobjInstruction(name='fc', ch='d{}'.format(i),
+                                                       t0=0, phase='-P1'),
+                                  PulseQobjInstruction(name='test_pulse_4',
+                                                       ch='d{}'.format(i), t0=0),
+                                  PulseQobjInstruction(name='fc', ch='d{}'.format(i),
+                                                       t0=0, phase='-P0')]),
+                Command(name='u3', qubits=[i],
+                        sequence=[PulseQobjInstruction(name='test_pulse_3',
+                                                       ch='d{}'.format(i), t0=0)])
+            ]
+
+        for connected_pair in self.cmap:
+            q1, q2 = connected_pair
+            cmd_def += [
+                Command(name='cx', qubits=[q1, q2],
+                        sequence=[PulseQobjInstruction(name='test_pulse_1',
+                                                       ch='d{}'.format(q1), t0=0),
+                                  PulseQobjInstruction(name='test_pulse_2',
+                                                       ch='u{}'.format(q1), t0=10),
+                                  PulseQobjInstruction(name='test_pulse_1',
+                                                       ch='d{}'.format(q2), t0=20),
+                                  PulseQobjInstruction(name='fc', ch='d{}'.format(q2),
+                                                       t0=20, phase=2.1)])
+            ]
+
+        return PulseDefaults(
+            qubit_freq_est=qubit_freq_est,
+            meas_freq_est=meas_freq_est,
+            buffer=buffer,
+            pulse_library=pulse_library,
+            cmd_def=cmd_def
+        )
+
+    def build_backend(self):
+        """Build backend."""
+        pass
+
+    def dump(self, folder: str):
+        """Dumps backend configuration files to specifier folder."""
+        with open('{0}/props_{1}.json'.format(folder, self.name), 'w') as f:
+            json.dump(self.build_props().to_dict(), f, indent=4, sort_keys=True)
+
+        with open('{0}/conf_{1}.json'.format(folder, self.name), 'w') as f:
+            json.dump(self.build_conf().to_dict(), f, indent=4, sort_keys=True)
+
+        with open('{0}/defs_{1}.json'.format(folder, self.name), 'w') as f:
+            json.dump(self.build_defaults().to_dict(), f,
+                      indent=4, sort_keys=True,
+                      default=lambda o: '')
+
+
+def fake_backend_generator(name: str,
```

Comment:
I think something is wrong here, perhaps this function isn't needed and `FakeBackendBuilder` could become `FakeBackend` which you could then instantiate with the desired parameters?